DUI3-DX/Converters/ArcGIS/Speckle.Converters.ArcGIS3/Utils/FeatureClassUtils.cs

```diff
@@ -0,0 +1,93 @@
+using ArcGIS.Core.Data;
+using ArcGIS.Core.Data.Exceptions;
+using ArcGIS.Core.Geometry;
+using Objects.GIS;
+using Speckle.Converters.Common;
+using Speckle.Converters.Common.Objects;
+using Speckle.Core.Models;
+
+namespace Speckle.Converters.ArcGIS3.Utils;
+
+public class FeatureClassUtils
+{
+  public void AddFeaturesToFeatureClass(
+    FeatureClass newFeatureClass,
+    VectorLayer target,
+    List<string> fieldAdded,
+    IRawConversion<Base, ArcGIS.Core.Geometry.Geometry> gisGeometryConverter
+  )
+  {
+    newFeatureClass.DeleteRows(new QueryFilter());
+    foreach (GisFeature feat in target.elements.Cast<GisFeature>())
+    {
+      using (RowBuffer rowBuffer = newFeatureClass.CreateRowBuffer())
+      {
+        foreach (string field in fieldAdded)
+        {
+          // try to assign values to writeable fields
+          try
+          {
+            if (feat.attributes is not null)
+            {
+              var value = feat.attributes[field];
+              if (value is not null)
+              {
+                rowBuffer[field] = value.ToString();
+              }
+              else
+              {
+                rowBuffer[field] = null;
+              }
+            }
+          }
+          catch (GeodatabaseFieldException)
+          {
+            // non-editable Field, do nothing
+          }
+        }
+
+        if (feat.geometry != null)
+        {
+          foreach (var geometryPart in feat.geometry)
+          {
+            // TODO: repeat for all geometries, add as Multipart
+            ArcGIS.Core.Geometry.Geometry nativeShape = gisGeometryConverter.RawConvert(geometryPart);
+            rowBuffer[newFeatureClass.GetDefinition().GetShapeField()] = nativeShape;
+            break;
+          }
+        }
+        newFeatureClass.CreateRow(rowBuffer).Dispose();
+      }
+    }
+  }
+
+  public GeometryType GetLayerGeometryType(VectorLayer target)
+  {
+    GeometryType geomType = new();
+    if (target.nativeGeomType == null)
+    {
+      throw new SpeckleConversionException($"Unknown geometry type for layer {target.name}");
+    }
+    else
+    {
+      if (target.nativeGeomType.ToLower().Contains("point"))
```

Comment:
is there a maybe cleaner pattern to this, can we predict reliably where pooint, polyline etc... will be in the string and look for this? probably not but thought I'd ask.

Reply:
relevant to the "primitive types" discussion, different host apps call these things differently, so far I haven't found a more accurate pattern for the purpose.

Reply:
Maybe we could look at the types, given this is taking the native ArcGIS type maybe there s something we can do

Reply:
these can come also from QGIS